.github/workflows/ci.yaml

```diff
@@ -22,6 +22,17 @@ jobs:
   lint:
     name: Lint
     runs-on: ubuntu-latest
+    strategy:
```

Comment:
why do you need all these to be different jobs? It looks a bit weird in the checks list

Reply:
It's meant to specify the jobs in a more concise way. I will change it back and just add the caching step.

Reply:
I've also changed the `actions/checkout` version. From v3 to v4. The `prover` job it's meant to be removed in a new PR.